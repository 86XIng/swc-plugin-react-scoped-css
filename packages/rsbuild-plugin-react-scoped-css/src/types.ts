export type PluginReactScopedCssOptions = {
  /**
   *
   * file include
   * @default /\.scoped\.(c|le|sc|sa)ss$/
   */
  include?: RegExp | string;
  /*
   * base hash seed
   */
  hashSeed?: string;
};
