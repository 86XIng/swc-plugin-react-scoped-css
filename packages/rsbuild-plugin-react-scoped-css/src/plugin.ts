import { mergeRsbuildConfig, type RsbuildPlugin } from '@rsbuild/core';
import { PluginReactScopedCssOptions } from './types';

export const pluginReactScopedCss = (opts: PluginReactScopedCssOptions = {}): RsbuildPlugin => {
  const options = opts ? {...opts} : {} as PluginReactScopedCssOptions;

  if (options?.include) {
    const type = Object.prototype.toString.call(options.include);
    if (type === '[object RegExp]') {
      options.include = options.include.toString().slice(1, -1);
    } else if (type !== '[object String]') {
      throw new Error('The type of include must be RegExp or String');
    }
  }

  return {
    name: 'scope:react-scoped-css',
    setup(api) {
      api.modifyRspackConfig((config) => {
        if (config.module?.rules) {
          for (let rule of config.module.rules) {
            if (rule && (rule as any).oneOf) {
              const oneOfs = (rule as any).oneOf;
              if (Array.isArray(oneOfs)) {
                for (let oneOf of oneOfs) {
                  if (oneOf.type === 'css' && Array.isArray(oneOf.use)) {
                    oneOf.use.unshift({
                      loader: require.resolve('scoped-css-loader'),
                    });
                  }
                }
              }
            }
          }
        }
  
        return config;
      });
  
      api.modifyRsbuildConfig((config) => {
        const newConfig = mergeRsbuildConfig(config, {
          tools: {
            swc: {
              jsc: {
                experimental: {
                  plugins: [
                    [
                      require.resolve('swc-plugin-react-scoped-css'),
                      options,
                    ],
                  ],
                },
              },
            },
          },
        });
        return newConfig;
      });
    },
  }
};
