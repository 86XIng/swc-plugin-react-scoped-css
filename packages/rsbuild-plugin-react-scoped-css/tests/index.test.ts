import { expect, describe, it } from 'vitest';
import { pluginReactScopedCss } from '../src';
import { createRsbuild } from '@rsbuild/core';

describe('plugins/react-scoped-css', () => {
  it('should apply react aliases by default', async () => {
    const rsbuild = await createRsbuild({
      rsbuildConfig: {
        plugins: [pluginReactScopedCss()]
      }
    });

    const configs = await rsbuild.initConfigs();
    expect(configs[0]).toMatchSnapshot();
  });
});
