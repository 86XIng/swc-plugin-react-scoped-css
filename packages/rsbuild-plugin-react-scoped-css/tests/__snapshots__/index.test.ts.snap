// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`plugins/react-scoped-css > should apply react aliases by default 1`] = `
{
  "context": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/packages/rsbuild-plugin-react-scoped-css",
  "devtool": "cheap-module-source-map",
  "entry": {
    "index": [
      "/Users/limbo/Workspace/swc-plugin-react-scoped-css/packages/rsbuild-plugin-react-scoped-css/src/index.ts",
    ],
  },
  "experiments": {
    "asyncWebAssembly": true,
  },
  "infrastructureLogging": {
    "level": "error",
  },
  "mode": "development",
  "module": {
    "generator": {
      "css/module": {
        "exportsConvention": "camel-case",
        "exportsOnly": false,
        "localIdentName": "[path][name]__[local]-[hash:6]",
      },
    },
    "parser": {
      "css/module": {
        "namedExports": false,
      },
      "javascript": {
        "exportsPresence": "error",
      },
    },
    "rules": [
      {
        "resolve": {
          "fullySpecified": false,
        },
        "test": /\\\\\\.m\\?js/,
      },
      {
        "include": [
          {
            "and": [
              "/Users/limbo/Workspace/swc-plugin-react-scoped-css/packages/rsbuild-plugin-react-scoped-css",
              {
                "not": /\\[\\\\\\\\/\\]node_modules\\[\\\\\\\\/\\]/,
              },
            ],
          },
          /\\\\\\.\\(\\?:ts\\|tsx\\|jsx\\|mts\\|cts\\)\\$/,
        ],
        "test": /\\\\\\.\\(\\?:js\\|jsx\\|mjs\\|cjs\\|ts\\|tsx\\|mts\\|cts\\)\\$/,
        "type": "javascript/auto",
        "use": [
          {
            "loader": "builtin:swc-loader",
            "options": {
              "env": {
                "coreJs": "3.36",
                "mode": "usage",
                "shippedProposals": true,
                "targets": [
                  "chrome >= 87",
                  "edge >= 88",
                  "firefox >= 78",
                  "safari >= 14",
                ],
              },
              "isModule": "unknown",
              "jsc": {
                "experimental": {
                  "plugins": [
                    [
                      "/Users/limbo/Workspace/swc-plugin-react-scoped-css/packages/swc-plugin-react-scoped-css/target/wasm32-wasi/release/swc_plugin_react_scoped_css.wasm",
                      {},
                    ],
                  ],
                },
                "externalHelpers": true,
                "parser": {
                  "decorators": true,
                  "syntax": "typescript",
                  "tsx": true,
                },
                "preserveAllComments": true,
                "transform": {
                  "decoratorMetadata": true,
                  "legacyDecorator": true,
                  "useDefineForClassFields": false,
                },
              },
              "sourceMaps": true,
            },
          },
        ],
      },
      {
        "mimetype": {
          "or": [
            "text/javascript",
            "application/javascript",
          ],
        },
        "resolve": {
          "fullySpecified": false,
        },
        "use": [
          {
            "loader": "builtin:swc-loader",
            "options": {
              "env": {
                "coreJs": "3.36",
                "mode": "usage",
                "shippedProposals": true,
                "targets": [
                  "chrome >= 87",
                  "edge >= 88",
                  "firefox >= 78",
                  "safari >= 14",
                ],
              },
              "isModule": "unknown",
              "jsc": {
                "experimental": {
                  "plugins": [
                    [
                      "/Users/limbo/Workspace/swc-plugin-react-scoped-css/packages/swc-plugin-react-scoped-css/target/wasm32-wasi/release/swc_plugin_react_scoped_css.wasm",
                      {},
                    ],
                  ],
                },
                "externalHelpers": true,
                "parser": {
                  "decorators": true,
                  "syntax": "typescript",
                  "tsx": true,
                },
                "preserveAllComments": true,
                "transform": {
                  "decoratorMetadata": true,
                  "legacyDecorator": true,
                  "useDefineForClassFields": false,
                },
              },
              "sourceMaps": true,
            },
          },
        ],
      },
      {
        "oneOf": [
          {
            "generator": {
              "filename": "static/image/[name].[contenthash:8][ext]",
            },
            "resourceQuery": /\\(__inline=false\\|url\\)/,
            "type": "asset/resource",
          },
          {
            "resourceQuery": /inline/,
            "type": "asset/inline",
          },
          {
            "generator": {
              "filename": "static/image/[name].[contenthash:8][ext]",
            },
            "parser": {
              "dataUrlCondition": {
                "maxSize": 10000,
              },
            },
            "type": "asset",
          },
        ],
        "test": /\\\\\\.\\(\\?:png\\|jpg\\|jpeg\\|pjpeg\\|pjp\\|gif\\|bmp\\|webp\\|ico\\|apng\\|avif\\|tif\\|tiff\\|jfif\\)\\$/i,
      },
      {
        "oneOf": [
          {
            "generator": {
              "filename": "static/svg/[name].[contenthash:8].svg",
            },
            "resourceQuery": /\\(__inline=false\\|url\\)/,
            "type": "asset/resource",
          },
          {
            "resourceQuery": /inline/,
            "type": "asset/inline",
          },
          {
            "generator": {
              "filename": "static/svg/[name].[contenthash:8].svg",
            },
            "parser": {
              "dataUrlCondition": {
                "maxSize": 10000,
              },
            },
            "type": "asset",
          },
        ],
        "test": /\\\\\\.svg\\$/i,
      },
      {
        "oneOf": [
          {
            "generator": {
              "filename": "static/media/[name].[contenthash:8][ext]",
            },
            "resourceQuery": /\\(__inline=false\\|url\\)/,
            "type": "asset/resource",
          },
          {
            "resourceQuery": /inline/,
            "type": "asset/inline",
          },
          {
            "generator": {
              "filename": "static/media/[name].[contenthash:8][ext]",
            },
            "parser": {
              "dataUrlCondition": {
                "maxSize": 10000,
              },
            },
            "type": "asset",
          },
        ],
        "test": /\\\\\\.\\(\\?:mp4\\|webm\\|ogg\\|mov\\|mp3\\|wav\\|flac\\|aac\\|m4a\\|opus\\)\\$/i,
      },
      {
        "oneOf": [
          {
            "generator": {
              "filename": "static/font/[name].[contenthash:8][ext]",
            },
            "resourceQuery": /\\(__inline=false\\|url\\)/,
            "type": "asset/resource",
          },
          {
            "resourceQuery": /inline/,
            "type": "asset/inline",
          },
          {
            "generator": {
              "filename": "static/font/[name].[contenthash:8][ext]",
            },
            "parser": {
              "dataUrlCondition": {
                "maxSize": 10000,
              },
            },
            "type": "asset",
          },
        ],
        "test": /\\\\\\.\\(\\?:woff\\|woff2\\|eot\\|ttf\\|otf\\|ttc\\)\\$/i,
      },
      {
        "dependency": "url",
        "generator": {
          "filename": "static/wasm/[hash].module.wasm",
        },
        "test": /\\\\\\.wasm\\$/,
        "type": "asset/resource",
      },
      {
        "oneOf": [
          {
            "resolve": {
              "preferRelative": true,
            },
            "sideEffects": true,
            "test": /\\\\\\.module\\\\\\.\\\\w\\+\\$/i,
            "type": "css/module",
            "use": [
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/scoped-css-loader@1.0.0/node_modules/scoped-css-loader/index.js",
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/postcss-loader",
                "options": {
                  "postcssOptions": {
                    "config": false,
                    "plugins": [
                      {
                        "browsers": [
                          "chrome >= 87",
                          "edge >= 88",
                          "firefox >= 78",
                          "safari >= 14",
                        ],
                        "info": [Function],
                        "options": {
                          "flexbox": "no-2009",
                          "overrideBrowserslist": [
                            "chrome >= 87",
                            "edge >= 88",
                            "firefox >= 78",
                            "safari >= 14",
                          ],
                        },
                        "postcssPlugin": "autoprefixer",
                        "prepare": [Function],
                      },
                    ],
                  },
                  "sourceMap": false,
                },
              },
            ],
          },
          {
            "resolve": {
              "preferRelative": true,
            },
            "sideEffects": true,
            "type": "css",
            "use": [
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/scoped-css-loader@1.0.0/node_modules/scoped-css-loader/index.js",
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/postcss-loader",
                "options": {
                  "postcssOptions": {
                    "config": false,
                    "plugins": [
                      {
                        "browsers": [
                          "chrome >= 87",
                          "edge >= 88",
                          "firefox >= 78",
                          "safari >= 14",
                        ],
                        "info": [Function],
                        "options": {
                          "flexbox": "no-2009",
                          "overrideBrowserslist": [
                            "chrome >= 87",
                            "edge >= 88",
                            "firefox >= 78",
                            "safari >= 14",
                          ],
                        },
                        "postcssPlugin": "autoprefixer",
                        "prepare": [Function],
                      },
                    ],
                  },
                  "sourceMap": false,
                },
              },
            ],
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      {
        "oneOf": [
          {
            "resolve": {
              "preferRelative": true,
            },
            "sideEffects": true,
            "test": /\\\\\\.module\\\\\\.\\\\w\\+\\$/i,
            "type": "css/module",
            "use": [
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/scoped-css-loader@1.0.0/node_modules/scoped-css-loader/index.js",
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/postcss-loader",
                "options": {
                  "postcssOptions": {
                    "config": false,
                    "plugins": [
                      {
                        "browsers": [
                          "chrome >= 87",
                          "edge >= 88",
                          "firefox >= 78",
                          "safari >= 14",
                        ],
                        "info": [Function],
                        "options": {
                          "flexbox": "no-2009",
                          "overrideBrowserslist": [
                            "chrome >= 87",
                            "edge >= 88",
                            "firefox >= 78",
                            "safari >= 14",
                          ],
                        },
                        "postcssPlugin": "autoprefixer",
                        "prepare": [Function],
                      },
                    ],
                  },
                  "sourceMap": false,
                },
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/less-loader",
                "options": {
                  "implementation": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/less",
                  "lessOptions": {
                    "javascriptEnabled": true,
                    "paths": [
                      "/Users/limbo/Workspace/swc-plugin-react-scoped-css/packages/rsbuild-plugin-react-scoped-css/node_modules",
                    ],
                  },
                  "sourceMap": false,
                },
              },
            ],
          },
          {
            "resolve": {
              "preferRelative": true,
            },
            "sideEffects": true,
            "type": "css",
            "use": [
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/scoped-css-loader@1.0.0/node_modules/scoped-css-loader/index.js",
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/postcss-loader",
                "options": {
                  "postcssOptions": {
                    "config": false,
                    "plugins": [
                      {
                        "browsers": [
                          "chrome >= 87",
                          "edge >= 88",
                          "firefox >= 78",
                          "safari >= 14",
                        ],
                        "info": [Function],
                        "options": {
                          "flexbox": "no-2009",
                          "overrideBrowserslist": [
                            "chrome >= 87",
                            "edge >= 88",
                            "firefox >= 78",
                            "safari >= 14",
                          ],
                        },
                        "postcssPlugin": "autoprefixer",
                        "prepare": [Function],
                      },
                    ],
                  },
                  "sourceMap": false,
                },
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/less-loader",
                "options": {
                  "implementation": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/less",
                  "lessOptions": {
                    "javascriptEnabled": true,
                    "paths": [
                      "/Users/limbo/Workspace/swc-plugin-react-scoped-css/packages/rsbuild-plugin-react-scoped-css/node_modules",
                    ],
                  },
                  "sourceMap": false,
                },
              },
            ],
          },
        ],
        "test": /\\\\\\.less\\$/,
      },
      {
        "oneOf": [
          {
            "resolve": {
              "preferRelative": true,
            },
            "sideEffects": true,
            "test": /\\\\\\.module\\\\\\.\\\\w\\+\\$/i,
            "type": "css/module",
            "use": [
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/scoped-css-loader@1.0.0/node_modules/scoped-css-loader/index.js",
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/postcss-loader",
                "options": {
                  "postcssOptions": {
                    "config": false,
                    "plugins": [
                      {
                        "browsers": [
                          "chrome >= 87",
                          "edge >= 88",
                          "firefox >= 78",
                          "safari >= 14",
                        ],
                        "info": [Function],
                        "options": {
                          "flexbox": "no-2009",
                          "overrideBrowserslist": [
                            "chrome >= 87",
                            "edge >= 88",
                            "firefox >= 78",
                            "safari >= 14",
                          ],
                        },
                        "postcssPlugin": "autoprefixer",
                        "prepare": [Function],
                      },
                    ],
                  },
                  "sourceMap": false,
                },
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/resolve-url-loader",
                "options": {
                  "join": [Function],
                  "sourceMap": false,
                },
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/sass-loader",
                "options": {
                  "implementation": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/sass",
                  "sourceMap": true,
                },
              },
            ],
          },
          {
            "resolve": {
              "preferRelative": true,
            },
            "sideEffects": true,
            "type": "css",
            "use": [
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/scoped-css-loader@1.0.0/node_modules/scoped-css-loader/index.js",
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/postcss-loader",
                "options": {
                  "postcssOptions": {
                    "config": false,
                    "plugins": [
                      {
                        "browsers": [
                          "chrome >= 87",
                          "edge >= 88",
                          "firefox >= 78",
                          "safari >= 14",
                        ],
                        "info": [Function],
                        "options": {
                          "flexbox": "no-2009",
                          "overrideBrowserslist": [
                            "chrome >= 87",
                            "edge >= 88",
                            "firefox >= 78",
                            "safari >= 14",
                          ],
                        },
                        "postcssPlugin": "autoprefixer",
                        "prepare": [Function],
                      },
                    ],
                  },
                  "sourceMap": false,
                },
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/resolve-url-loader",
                "options": {
                  "join": [Function],
                  "sourceMap": false,
                },
              },
              {
                "loader": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/sass-loader",
                "options": {
                  "implementation": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+shared@0.6.1_@swc+helpers@0.5.3/node_modules/@rsbuild/shared/compiled/sass",
                  "sourceMap": true,
                },
              },
            ],
          },
        ],
        "test": /\\\\\\.s\\(\\?:a\\|c\\)ss\\$/,
      },
    ],
  },
  "name": "Client",
  "optimization": {
    "minimize": false,
    "splitChunks": {
      "cacheGroups": {
        "lib-axios": {
          "name": "lib-axios",
          "priority": 0,
          "reuseExistingChunk": true,
          "test": /\\[\\\\\\\\/\\]node_modules\\[\\\\\\\\/\\]\\(axios\\|axios-\\.\\+\\)\\[\\\\\\\\/\\]/,
        },
        "lib-lodash": {
          "name": "lib-lodash",
          "priority": 0,
          "reuseExistingChunk": true,
          "test": /\\[\\\\\\\\/\\]node_modules\\[\\\\\\\\/\\]\\(lodash\\|lodash-es\\)\\[\\\\\\\\/\\]/,
        },
        "lib-polyfill": {
          "name": "lib-polyfill",
          "priority": 0,
          "reuseExistingChunk": true,
          "test": /\\[\\\\\\\\/\\]node_modules\\[\\\\\\\\/\\]\\(tslib\\|core-js\\|@babel\\\\/runtime\\|@swc\\\\/helpers\\)\\[\\\\\\\\/\\]/,
        },
      },
      "chunks": "all",
      "enforceSizeThreshold": 50000,
    },
  },
  "output": {
    "chunkFilename": "static/js/async/[name].js",
    "cssChunkFilename": "static/css/async/[name].css",
    "cssFilename": "static/css/[name].css",
    "filename": "static/js/[name].js",
    "hashFunction": "xxhash64",
    "path": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/packages/rsbuild-plugin-react-scoped-css/dist",
    "pathinfo": false,
    "publicPath": "/",
    "webassemblyModuleFilename": "static/wasm/[hash].module.wasm",
  },
  "performance": {
    "hints": false,
  },
  "plugins": [
    HotModuleReplacementPlugin {
      "name": "HotModuleReplacementPlugin",
    },
    HtmlWebpackPlugin {
      "options": {
        "base": false,
        "cache": true,
        "chunks": [
          "index",
        ],
        "chunksSortMode": "auto",
        "compile": true,
        "entryName": "index",
        "excludeChunks": [],
        "favicon": false,
        "filename": "index.html",
        "hash": false,
        "inject": "head",
        "meta": {
          "charset": {
            "charset": "UTF-8",
          },
          "viewport": "width=device-width, initial-scale=1.0",
        },
        "minify": false,
        "publicPath": "auto",
        "scriptLoading": "defer",
        "showErrors": true,
        "template": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+core@0.6.1/node_modules/@rsbuild/core/static/template.html",
        "templateContent": false,
        "templateParameters": [Function],
        "title": "Rsbuild App",
        "xhtml": false,
      },
      "userOptions": {
        "chunks": [
          "index",
        ],
        "entryName": "index",
        "filename": "index.html",
        "inject": "head",
        "meta": {
          "charset": {
            "charset": "UTF-8",
          },
          "viewport": "width=device-width, initial-scale=1.0",
        },
        "minify": false,
        "scriptLoading": "defer",
        "template": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@rsbuild+core@0.6.1/node_modules/@rsbuild/core/static/template.html",
        "templateParameters": [Function],
        "title": "Rsbuild App",
      },
      "version": 5,
    },
    HtmlBasicPlugin {
      "name": "HtmlBasicPlugin",
      "options": {
        "index": {},
      },
    },
    DefinePlugin {
      "_options": {
        "process.env.ASSET_PREFIX": """",
        "process.env.NODE_ENV": ""test"",
      },
      "affectedHooks": "compilation",
      "name": "DefinePlugin",
    },
    RemoveCssSourcemapPlugin {
      "name": "RemoveCssSourcemapPlugin",
    },
  ],
  "resolve": {
    "alias": {
      "@swc/helpers": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/@swc+helpers@0.5.3/node_modules/@swc/helpers",
      "core-js": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/node_modules/.pnpm/core-js@3.36.1/node_modules/core-js",
    },
    "extensions": [
      ".ts",
      ".tsx",
      ".js",
      ".jsx",
      ".mjs",
      ".json",
    ],
    "tsConfigPath": "/Users/limbo/Workspace/swc-plugin-react-scoped-css/packages/rsbuild-plugin-react-scoped-css/tsconfig.json",
  },
  "target": [
    "web",
    "es2017",
  ],
}
`;
